throughput/processor.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package throughput
+
+import (
+	"context"
+	"encoding/binary"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ava-labs/hypersdk/api/ws"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/utils"
+)
+
+type txProcessor struct {
+	issuerWg sync.WaitGroup
+	inflight atomic.Int64
+
+	l            sync.Mutex
+	confirmedTxs int
+	totalTxs     int
+
+	sent atomic.Int64
+}
+
+func (tp *txProcessor) logResult(
+	result *chain.Result,
+	wsErr error,
+) {
+	tp.l.Lock()
+	if result != nil {
+		if result.Success {
+			tp.confirmedTxs++
+		} else {
+			utils.Outf("{{orange}}on-chain tx failure:{{/}} %s %t\n", string(result.Error), result.Success)
+		}
+	} else {
+		// We can't error match here because we receive it over the wire.
+		if !strings.Contains(wsErr.Error(), ws.ErrExpired.Error()) {
+			utils.Outf("{{orange}}pre-execute tx failure:{{/}} %v\n", wsErr)
+		}
+	}
+	tp.totalTxs++
+	tp.l.Unlock()
+}
+
+func (tp *txProcessor) logIssuerState(ctx context.Context, issuer *issuer) {
```

Comment:
could we switch from passing the whole `issuer` in to including a reference to a client that the tx processor can use to fetch unit prices?